port/unix/omrsysinfo.c

```diff
@@ -3459,38 +3542,68 @@ omrsysinfo_cgroup_is_limits_supported(struct OMRPortLibrary *portLibrary)
 			goto _end;
 		}
 
+		omrthread_monitor_enter(cgroupEntryListMonitor);
 		rc = readCgroupFile(portLibrary, getpid(), &PPG_cgroupEntryList);
+		omrthread_monitor_exit(cgroupEntryListMonitor);
+		if (0 != rc) {
+			goto _end;
+		}
+		PPG_cgroupSubsystemsAvailable = getAvailableSubsystems(portLibrary, PPG_cgroupEntryList);
```

Comment:
Once readCgroupFile() has set PPG_cgroupEntryList to non-null, the PPG_cgroupSubsystemsAvailable global isn't read and this function will always return 0.

It seems strange that this function calls `getAvailableSubsystems`.  I would have (maybe naively?) expected that to be called by `omrsysinfo_cgroup_get_available_subsystems` if the `PPG_cgroupSubsystemsAvailable` global was null...

Reply:
`PPG_cgroupSubsystemsAvailable` cannot be checked for null; its not a pointer.
`omrsysinfo_cgroup_is_system_available` calls `getAvailableSubsystems` to ensure `PPG_cgroupSubsystemsAvailable` is set properly at the time when `PPG_cgroupEntryList` is created.
If `getAvailableSubsystems` is to be called by `omrsysinfo_cgroup_get_available_subsystems`, then it cannot rely on `PPG_cgroupEntryList` being null to call it, as `PPG_cgroupEntryList` may have been set previously by a call to `omrsysinfo_cgroup_is_system_available`.